### ISSUE TITLE
Provide Field/Part Definition in GraphQl

### ISSUE THREAD

**Issue by deanmarcussen** [OrchardCMS/OrchardCore#6236]
It would be useful to provide the Field or Part Definition in GraphQl

This is helpful when access to the field settings is required, example here
https://github.com/OrchardCMS/OrchardCore/pull/6016/files#diff-f626f24883701a8be1900dbd89142848R54-R57

Probably it would need to change to `ResolveFieldContext<MarkdownField>` to be more like
`ResolveFieldContext<DisplayContext<MarkdownField>>`, where `DisplayContext<T>` is a wrapper, as the field and part definitions are available when building the field definitions.

(I did try to see if it was possible to use the `Metadata` property, but that is fixed at time of creating the schema, so didn't seem so useful)

/cc @carlwoodhouse

**Comment by carlwoodhouse** [comment]
Not sure exactly what you’re trying to achieve .. you need the settings to
know how to render the field or so ? Or ..

Just trying to grasp the use-case

**Comment by deanmarcussen** [comment]
Yes, in this case (and it was an edge case) it was about needing the settings for the `MarkdownField` to know whether to sanitize the render html output in `GraphQl` or not.

I was able to get it out of the `JsonPath`, but longer term it would be better if it was less hacky ;)

**Comment by carlwoodhouse** [comment]
Just want to be clear (I know this is an edge case anyway) ...

But you want to know if you the consumer should sanitize it or not based on
the setting; or should graphql be returning it sanitized or not based on
the setting :)

**Comment by deanmarcussen** [comment]
In this case it's graphql returning it sanitized or not, because we have the `ToHtml` rendering option for `Markdown` in GraphQl.

So it's not about providing more metadata/settings to the consumer, just to the `RenderFieldContext` :)

(all to save this ... )
```
            var jObject = ctx.Source.Content as JObject;
            // The JObject.Path is consistent here even when contained in a bag part.
            var jsonPath = jObject.Path;
            var paths = jsonPath.Split('.');
            var partName = paths[0];
            var fieldName = paths[1];
```

**Comment by carlwoodhouse** [comment]
Maybe we should just have an argument on the field to sanitize or not
regardless of the setting and leave it to the consumer lol ..

**Comment by carlwoodhouse** [comment]
If the settings only change per definition though Not per item; you could
technically use metadata like you said ... the schema should be regenerated
on definition change,

**Comment by deanmarcussen** [comment]
Yeah let me have a look again there was something that didn’t seem right about the metadata property but worth another look (def easier)
